### PR TITLE
hardening/564_performance_doc_add_batching

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #564 
* No tests are required
* Assignee @fgalan

### CODE REVIEW

File: doc/operation/performance_tuning_tips.md [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
+
+Obviously, not all the events will always regard to the same unique entity, and many entities may be involved within a batch. But that's not a problem, since several sub-batches of events are created within a batch, one sub-batch per final destination HDFS file. In the worst case, the whole 100 entities will be about 100 different entities (100 different HDFS destinations), but that will not be the usual scenario. Thus, assuming a realistic number of 10-15 sub-batches per batch, we are replacing the 100 writes of the event by event approach with only 10-15 writes.
+
+Nevertheless, a risk when using batches is the last batch never gets built. I.e. in the above 100 size batch if only 99 events are notified and the 100th event never arrives, then the batch is never ready to be preocessed by the sink. Thats the reason the batch mechanism adds an accumulation timeout to prevent the sink stays in an eternal state of batch building when no new data arrives. If such a timeout is reached, then the batch is persisted as it is.
```
Comment (fgalan) [inline doc/operation/performance_tuning_tips.md]: There is another risk (that I think that it is even worse than the one related with persisting delays): it introduces state in Cygnus, so if Cygnus crashes or it is stoped, then all the accumulated batch is lost. I think that also should be mentioned in the documentation.

File: doc/operation/performance_tuning_tips.md [hunk 2]
```diff
@@ -1,6 +1,40 @@
...
+
+By default, all the sinks have a configured batch size and batch accumulation timeout of 1 and 30 seconds, respectively. These are the parameters all the sinks have for these purpose:
+
+    <agent_name>.sinks.<sink_name>.batch_size = 1
+    <agent_name>.sinks.<sink_name>.batch_timeout = 30
+
+Nevertheless, as explained above, it is highly recommended to increase at least the batch size for performance purposes. Which are the optimal values? The size of the batch it is closely related to the transaction size of the channel the events are got from (it has no sense the first one is greater then the second one), and it depends on the number of estimated sub-batches as well. The accumulation timeout will depend on how often you want to see new data in the final storage.
+
```
Comment (fgalan) [inline doc/operation/performance_tuning_tips.md]: As final note (maybe already mentioned but I have missed it?) it should be mention that not all technologies allow batching (and probably an enumeration of current sinks that wouldn't benefit performance when using batching, e.g. not remember now if MySQL was one of the cases).

File: doc/operation/performance_tuning_tips.md [hunk 3]
```diff
@@ -100,19 +138,24 @@ Due to the available <i>Channel Selectors</i> do not fit our needs, a custom sel
...
-* Subset of channels associated to each storage. The union of all the subsets must be equal to all the channels configured for the source. E.g. if `cygnusagent.sources.mysource.channels = ch1 ch2 ch3 ch4 ch5 ch6` and if `ch1` is associated to a MySQL storage, `ch2` and `ch3` are associated to a CKAN storage and `ch4`, `ch5` and `ch6` are associated to a HDFS storage then `cygnusagent.sources.mysource.selector.storages.storage1 = ch1`, `cygnusagent.sources.mysource.selector.storages.storage2 = ch2,ch3` and `cygnusagent.sources.mysource.selector.storages.storage3 = ch4,ch5,ch6`. 
+* Subset of channels associated to each storage. The union of all the subsets must be equal to all the channels configured for the source. E.g. if `cygnusagent.sources.mysource.channels = ch1 ch2 ch3 ch4 ch5 ch6` and if `ch1` is associated to a MySQL storage, `ch2` and `ch3` are associated to a CKAN storage and `ch4`, `ch5` and `ch6` are associated to a HDFS storage then `cygnusagent.sources.mysource.selector.storages.storage1 = ch1`, `cygnusagent.sources.mysource.selector.storages.storage2 = ch2,ch3` and `cygnusagent.sources.mysource.selector.storages.storage3 = ch4,ch5,ch6`.
```
Comment (fgalan) [inline doc/operation/performance_tuning_tips.md]: Very long line... which are the actual changes?

File: doc/operation/performance_tuning_tips.md [hunk 4]
```diff
@@ -123,22 +166,47 @@ In order to calculate the appropiate capacity, just have in consideration the fo
...
-Fermín Galán Márquez (fermin.galanmarquez@telefonica.com) 
\ No newline at end of file
+[Top](#top)
+
+##<a name="section6"></a>Writing logs
+Writing logs, as any I/O operation where disk writes are involved, is largely slow. Please avoid writing a huge number if logs unless necessary, i.e. because your are debuging Cygnus, and try running cygnus at least with `INFO` level (despite a lot of logs are still written at that level). The best is running with `ERROR` level.
+
+Logging level Cygnus run with is configured in `/usr/cugnus/conf/log4.properties`. `INFO` is configured by dafault:
+
+     flume.root.logger=INFO,LOGFILE
```
Comment (fgalan) [inline doc/operation/performance_tuning_tips.md]: Is there any way of disabling logging at all?

File: doc/operation/performance_tuning_tips.md (hunk 2 above)
Comment (frbattid) [inline doc/operation/performance_tuning_tips.md]: Such a note is at the begining of the section ;)

File: doc/operation/performance_tuning_tips.md (hunk 3 above)
Comment (frbattid) [inline doc/operation/performance_tuning_tips.md]: There were no changes form my side in this section. Maybe a line feed or something like that making Github gets crazy.

File: doc/operation/performance_tuning_tips.md (hunk 4 above)
Comment (frbattid) [inline doc/operation/performance_tuning_tips.md]: Yes, with `OFF`. I will detail it.

File: doc/operation/performance_tuning_tips.md (hunk 1 above)
Comment (frbattid) [inline doc/operation/performance_tuning_tips.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/7b5b9840fcb6deefeecfc49393828e3a0ba3fb31

File: doc/operation/performance_tuning_tips.md (hunk 4 above)
Comment (frbattid) [inline doc/operation/performance_tuning_tips.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/7b5b9840fcb6deefeecfc49393828e3a0ba3fb31

File: doc/operation/performance_tuning_tips.md (hunk 3 above)
Comment (fgalan) [inline doc/operation/performance_tuning_tips.md]: NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**